src/Gui/FolderConverterTab.xaml.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Avalonia.Controls;
+using Avalonia.Interactivity;
+using Avalonia.Markup.Xaml;
+using Avalonia.Threading;
+
+namespace Pytocs.Gui
+{
+    public class FolderConverterTab : UserControl
+    {
+        private object SyncRoot { get; } = new object();
+
+        private bool _isConversionInProgress;
+
+        private TextBox TargetFolderBox { get; }
+
+        private TextBox SourceFolderBox { get; }
+
+        private TextBox ConversionLogBox { get; }
+
+        private Func<string, Task> AppendLog { get; }
+
+        public FolderConverterTab()
+        {
+            this.InitializeComponent();
+            SourceFolderBox = this.FindControl<TextBox>(nameof(SourceFolderBox));
+            TargetFolderBox = this.FindControl<TextBox>(nameof(TargetFolderBox));
+            ConversionLogBox = this.FindControl<TextBox>(nameof(ConversionLogBox));
+
+            AppendLog = x => Dispatcher.UIThread.InvokeAsync(() => ConversionLogBox.Text += x);
+        }
+
+        private void InitializeComponent()
+        {
+            AvaloniaXamlLoader.Load(this);
+        }
+
+        private async void BrowseSource_Click(object sender, RoutedEventArgs e)
+        {
+            var dialog = new OpenFolderDialog();
+            var result = await dialog.ShowAsync();
+
+            if (!string.IsNullOrWhiteSpace(result))
+            {
+                SourceFolderBox.Text = result;
+            }
+        }
+
+        private async void BrowseTarget_Click(object sender, RoutedEventArgs e)
+        {
+            var dialog = new OpenFolderDialog();
+            var result = await dialog.ShowAsync();
+
+            if (!string.IsNullOrWhiteSpace(result))
+            {
+                TargetFolderBox.Text = result;
+            }
+        }
+
+        private async void Convert_Click(object sender, RoutedEventArgs e)
+        {
+            var (sourceFolder, targetFolder) = GetValidPaths();
+
+            if (sourceFolder == null)
+            {
+                return;
+            }
+
+            lock (SyncRoot)
+            {
+                if (_isConversionInProgress)
+                {
+                    return;
+                }
+
+                _isConversionInProgress = true;
+            }
+
+            ConversionLogBox.Text = string.Empty;
+
```

Comment:
You should probably disable all controls until the async operation completes. That makes the `syncroot` and `_isConversionInProgress` unnecessary. 

Consider also adding, in a separate PR, a `Cancel` button to cancel a long-running conversion. You could also change the `Text` property of the `Convert` button to `Cancel` during the conversion process, then changing it back to `Convert` once conversion is done.